lib/formatters/sarif.js

```diff
@@ -86,8 +87,24 @@ class SarifPrinter {
           }
         }
 
-        if (result.messages.length > 0) {
-          for (const message of result.messages) {
+        let fileErrors = result.messages;
+        let filteredMessages;
+
+        if (fileErrors.length > 0) {
+          let errorsFiltered = fileErrors.filter(
+            (error) => error.severity === Linter.ERROR_SEVERITY
+          );
+          let warnings = this.options.quiet
+            ? []
+            : fileErrors.filter((error) => error.severity === Linter.WARNING_SEVERITY);
+          let todos =
+            this.options.quiet || !this.options.includeTodo
```

Comment:
Why is this an `||`? I would think that if you have specified `--include-todo --quiet` you should still get todos included...

Reply:
I think quiet implies only errors, no?

Reply:
We're going to follow up with this in a later PR.

Reply:
Ya, I agree that is what it does in `--json` but I don't particularly like it. I'll open a separate issue to track that though since we should keep them in sync together either way.